components/admin_console/saml_settings.jsx

```diff
@@ -395,15 +416,10 @@ export default class SamlSettings extends AdminSettings {
                         />
                     }
                     placeholder={Utils.localizeMessage('admin.saml.assertionConsumerServiceURLEx', 'Ex "https://<your-mattermost-url>/login/sso/saml"')}
-                    helpText={
-                        <FormattedMessage
-                            id='admin.saml.assertionConsumerServiceURLDesc'
-                            defaultMessage='Enter https://<your-mattermost-url>/login/sso/saml. Make sure you use HTTP or HTTPS in your URL depending on your server configuration. This field is also known as the Assertion Consumer Service URL.'
-                        />
-                    }
+                    helpText={consumerServiceUrlHelp}
                     value={this.state.assertionConsumerServiceURL}
                     onChange={this.handleChange}
-                    disabled={!this.state.enable || !this.state.verify}
+                    disabled={!this.state.enable || !this.state.verify || this.state.siteUrlSet}
                 />
```

Comment:
So this field won’t be editable if we have the site Url set?

What happens to the actual setting when siteURL is modified?

Reply:
Yes according to the issue opened for this ticket, the setting should be non-editable if the Site URL is set. If site URL is modified, the service provider URL should change accordingly (Refer to line 54)

Reply:
Ok if it is non-editable the problem that I have is this.

1. Setup siteURL and save settings
2. go to SAML and edit the settings accordingly and save settings (here it will save the setting to the config file with the siteURL)
3. Update siteURL and save settings.

After step 3 if I don't go back to SAML settings then it won't be updated in the config.json file

Reply:
Fair point, should it be changed so that when the Site URL is updated and the service URL has been set previously, it updates the service URL as well?

Reply:
I would think so.. but perhaps we should wait until a PM takes a look at this